lib/cn/csched_sp3_work.c

```diff
@@ -651,28 +658,33 @@ sp3_work(
         if (action == CN_ACTION_SPLIT) {
             struct cn_tree_node *root = tree->ct_root;
 
-            /* Atomically increment the split/sync counters to prevent new compaction
-             * jobs from starting in both this node and the root node (does not apply
-             * to split jobs).  Once all root jobs finish a split job will be able to
-             * run with exclusive access to this node.
+            /* If there are no splits pending and root spill is behind
+             * then wait for throttling to engage and root spill to catch
+             * up before requesting a split.
              */
-            (*wp)->cw_resched = true;
-            root->tn_spillsync++;
-            tn->tn_spillsync++;
-
-            if (atomic_read(&root->tn_busycnt) > 0)
-                goto locked_nowork;
+            if (root->tn_rspill_sync == 0 &&
+                cn_ns_kvsets(&root->tn_ns) > thresh->rspill_runlen_min * 3) {
 
-            if (!cn_node_comp_token_get(root))
+                (*wp)->cw_resched = true;
+                ev_debug(1);
                 goto locked_nowork;
+            }
```

Comment:
We'll want to keep an eye on this with slow media. If with slow media rspill can't keep up then probably it indicates a problem with the throttle.  Hence the event counter for now...